ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderStateImpl.java

```diff
@@ -685,6 +687,22 @@ private synchronized void sendStartLeaderElectionToHigherPriorityPeer(RaftPeerId
     });
   }
 
+  boolean sendStartLeaderElection(FollowerInfo followerInfo) {
+    final RaftPeerId followerId = followerInfo.getPeer().getId();
+    final TermIndex leaderLastEntry = server.getState().getLastEntry();
+    if (leaderLastEntry == null) {
+      sendStartLeaderElection(followerId, null);
+      return true;
+    }
+
+    final long followerMatchIndex = followerInfo.getMatchIndex();
+    if (followerMatchIndex >= leaderLastEntry.getIndex()) {
```

Comment:
Same, I don't think `followerMatchIndex` can be greater than `leaderLastEntry.getIndex()`.
But it's okay to check just in case.

Reply:
We do not hold a lock so that the `matchIndex` could (theoretically) be updated after `getLastEntry()`.  I believe it won't happen in the current code though.